### DISCUSSION TITLE
Add a special function that evaluates nicely with undefined arguments.

### DISCUSSION THREAD

**Discussion by gusega** [open-policy-agent/community#561] [OPA and Rego] [answered]
Hi 

I have the following policy:

```rego
allow if count(errs) == 0
valid_document if {
...
}
errs contains sprintf("document must be %v but is %v", ["expected", input.document.something]) if {
    not valid_document
}
```
Unfortuntaly, if `input.document.something` is undefined, then `errs` is empty.

Could you please add a function that allows passing undefined documents and does something nice with them, like replace their value with `<undefined>` or something like that.

For example:
if `input.document.something` is undefined then
```rego
undef.sprintf("document must be %v but is %v", ["expected", input.document.something])
```
 returns `document must be expected but is <undefined>`

Thank you.

**Comment by anderseknert** [comment, marked as answer]
Hi there! There are a few ways you can achieve that with existing language constructs, but if you're looking for a built-in function specifically, `object.get` would be one to reach for here.

```rego
errs contains sprintf("document must be %v but is %v", ["expected", something]) if {
    not valid_document
    something := object.get(input, ["document", "something"], "<undefined>")
}
```
